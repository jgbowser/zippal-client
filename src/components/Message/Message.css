.Message_Form {
  margin-top: 15px;
  padding: 5px;
}

.Message__header {
  margin-bottom: 20px;
}

.Message__textarea {
  width: 100%;
  max-width: 400px;
  min-height: 100px;
}