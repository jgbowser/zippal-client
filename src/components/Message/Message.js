import React, { Component } from 'react';
import MessageService from '../../services/message-api-service';
import UserContext from '../../context/UserContext';
import './Message.css';

class Message extends Component {

    constructor(props) {
        super(props)
        this.state = {
            pending_content: 'Message in Progress...'
        }
        }

    static contextType = UserContext;

    componentDidMount = () => {
        this.setState({
            pending_content: this.props.message.content
        })
    }

    saveMessageHandler = async () => {
        const { pending_content } = this.state
        const message = await MessageService.saveMessage(pending_content, this.props.message.id)
        this.props.setNewMessage(message)
        if(this.props.clearSelectedMessage) {
        this.props.clearSelectedMessage()
        } else {
        this.props.closeNewConvoMessage()
        }
    }

    sendMessageHandler = async () => {
        const { pending_content } = this.state
        const message = await MessageService.sendMessage(pending_content, this.props.message.id)
        this.props.setNewMessage(message)
        if(this.props.clearSelectedMessage) {
        this.props.clearSelectedMessage()
        } else {
        this.props.closeNewConvoMessage()
        }
    }

    setPendingContent = (e) => {
        this.setState({
        pending_content: e.value
        })
    }

    renderMessageForm() {
        return (
        <form className='Message_Form'>
            <div className='Message_Container'>
            <label htmlFor='Message_Content'>
                Create a new message for {this.props.convoData.pal_name}
            </label>
            <textarea
                name='Message_Content_Area'
                type='text'
                id='Message_Content'
                value={this.state.pending_content}
                onChange={(e) => this.setPendingContent(e.target)}
            />
            </div>
            <div className='Message_Buttons_Container'>
            <button
                type='button'
                className='Message_Save_Btn'
                onClick={this.saveMessageHandler}
            >
                Save Message
                    </button>
            <button
                type='button'
                className='Message_Send_Btn'
                onClick={this.sendMessageHandler}
            >
                Send Message to {this.props.convoData.pal_name}
            </button>
            </div>
        </form>
        )
    }

    renderMessage() {
        return (
        <div className='Message_Read'>
            <h3> Message {
            this.context.user.id === this.props.message.sender_id
                ? 'to '
                : 'from '
            }
            {this.props.convoData.pal_name}
            </h3>
            <p>
            {this.props.message.content}
            </p>
        </div>
        )
    }

    render() {
        return this.props.message.sender_status === 'Sent' || this.props.message.sender_status === 'Awaiting Reply'
        ? this.renderMessage()
        : this.renderMessageForm()
    }
}

export default Message;