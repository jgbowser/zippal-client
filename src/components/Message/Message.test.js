import React from 'react'
import ReactDOM from 'react-dom'
import Message from './Message'
import testHelpers from '../../services/test-helpers'

describe('Message Component', () => {
  const convos = testHelpers.makeTestConvos()
  const convoData = convos[0]
  const messages = testHelpers.makeTestMessages()
  const message = messages[0]
  it('renders without crashing', () => {
    const user = {}
    const div = document.createElement('div')
    ReactDOM.render(<Message convoData={convoData} message={message} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})