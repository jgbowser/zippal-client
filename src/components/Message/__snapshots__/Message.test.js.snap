// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component takes a snapshot of Message componenet 1`] = `
<form
  className="Message_Form"
>
  <div
    className="Message_Container"
  >
    <label
      htmlFor="Message_Content"
    >
      Create a new message for 
      Test User
    </label>
    <textarea
      id="Message_Content"
      name="Message_Content_Area"
      onChange={[Function]}
      type="text"
      value="Sup"
    />
  </div>
  <div
    className="Message_Buttons_Container"
  >
    <button
      className="Message_Save_Btn"
      onClick={[Function]}
      type="button"
    >
      Save Message
    </button>
    <button
      className="Message_Send_Btn"
      onClick={[Function]}
      type="button"
    >
      Send Message to 
      Test User
    </button>
  </div>
</form>
`;
